src/actions/ScanActions.js

```diff
@@ -286,3 +302,63 @@ export const privateKeyModalActivated = () => async (dispatch: Dispatch, getStat
     )
   }, 1000)
 }
+
+const shownWalletGetCryptoModals = []
+
+export const checkAndShowGetCryptoModal = () => async (dispatch: Dispatch, getState: GetState) => {
+  try {
+    const state = getState()
+    const currencyCode = UI_SELECTORS.getSelectedCurrencyCode(state)
+    const wallet = UI_SELECTORS.getSelectedWallet(state)
+    // check if balance is zero
+    const balance = wallet.nativeBalances[currencyCode]
+    if (balance !== '0' || shownWalletGetCryptoModals.includes(wallet.id)) return // if there's a balance then early exit
+    shownWalletGetCryptoModals.push(wallet.id) // add to list of wallets with modal shown this session
+    let threeButtonModal
+    if (SPECIAL_CURRENCY_INFO[currencyCode].displayBuyCrypto) {
```

Comment:
This will crash if you get a currencyCode that isn't in the SPECIAL_CURRENCY_INFO array. use getSpecialCurrencyInfo() instead which is safe as it always returns an object.

Reply:
Ok